Tasks/TwineAuthenticateV1/twineauthenticatemain.ts

```diff
@@ -47,14 +41,58 @@ async function main(): Promise<void> {
         let pypircPath = utils.getPypircPath();
 
         // create new file. We do not merge existing files and always create a fresh file
-        fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
-        tl.setVariable("PYPIRC_PATH", pypircPath, false);
-        tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        if (!tl.getVariable("PYPIRC_PATH") || !tl.exist(tl.getVariable("PYPIRC_PATH"))) {
+            fs.writeFileSync(pypircPath, formPypircFormatFromData(newEndpointsToAdd));
+            tl.setVariable("PYPIRC_PATH", pypircPath, false);
+            tl.debug(tl.loc("VariableSetForPypirc", pypircPath));
+        }
+        else {
+            const pypirc = fs.readFileSync(tl.getVariable("PYPIRC_PATH"), 'utf8');
+            let fileContent = ini.parse(pypirc);
+
+            let dotRepos: string[] = [];
+
+            for (let entry of newEndpointsToAdd) {
+
+                tl.debug(tl.loc("Info_AddingAuthForRegistry", entry.packageSource.feedName));
+
+                if (entry.packageSource.feedName in fileContent){
+                    tl.debug(tl.loc("DuplicateRegistry", entry.packageSource.feedName));
+                    if (internalFeed.includes(entry)) {
+                        internalFeed.pop();
+                        continue;
+                    }
+                    externalEndpoints.pop();
+                    continue;
+                }
+
+                if (entry.packageSource.feedName.includes('.')) {
+                    let repo = `${os.EOL}[${entry.packageSource.feedName}]${os.EOL}repository=${entry.packageSource.feedUri}${os.EOL}username=${entry.username}${os.EOL}password=${entry.password}`;
```

Comment:
IS this the feedname parsing fix? can we separate this into a helper method? Are there any other characters allowed in the feed name that would also need to be escaped?

Reply:
That was the parsing fix for special characters. I ended up changing back to something very similar to what was in place before. To make things easier, I'm also using this for all feeds now so that we don't have to use an if-else statement.